sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12-file-detail.ts

```diff
@@ -1,114 +1,578 @@
-      record.append(this.applicationNumber, 10);
-      record.append(this.sin, 9);
-      record.appendWithEndFiller(this.studentLastName, 25, SPACE_FILLER);
-      record.appendWithEndFiller(this.studentGivenName ?? "", 15, SPACE_FILLER);
-      record.appendDate(this.birthDate, DATE_FORMAT);
-      record.appendWithEndFiller(this.programName, 25, SPACE_FILLER);
-      record.appendWithEndFiller(this.programDescription, 50, SPACE_FILLER);
-      record.appendWithEndFiller(this.credentialType, 25, SPACE_FILLER);
-      record.append(this.cipCode.toString(), 6);
-      record.append(this.nocCode, 4);
-      record.appendWithEndFiller(this.sabcCode ?? "", 4, SPACE_FILLER);
-      record.appendWithEndFiller(
-        this.institutionProgramCode ?? "",
-        25,
-        SPACE_FILLER,
-      );
-      record.append(this.programLength.toString(), 1);
-      record.appendDate(this.studyStartDate, DATE_FORMAT);
-      record.appendDate(this.studyEndDate, DATE_FORMAT);
-      record.appendWithStartFiller(
-        this.tuitionFees.toString(),
-        8,
-        NUMBER_FILLER,
-      );
-      record.appendWithStartFiller(
-        this.programRelatedCosts.toString(),
-        8,
-        NUMBER_FILLER,
-      );
-      record.appendWithStartFiller(
-        this.mandatoryFees.toString(),
-        8,
-        NUMBER_FILLER,
-      );
-      record.appendWithStartFiller(
-        this.exceptionExpenses.toString(),
-        8,
-        NUMBER_FILLER,
-      );
-      record.appendWithStartFiller(
-        this.totalFundedWeeks.toString(),
-        2,
-        NUMBER_FILLER,
-      );
-      record.repeatAppend(SPACE_FILLER, 3); //We have hardcoded the courseLoad to null as its only for FullTime.
-      record.append("F", 1); //This implementation is only for FullTime, so hardcoding it to F.
-      record.appendWithEndFiller(disbursement.coeStatus, 10, SPACE_FILLER);
-      record.appendWithEndFiller(
-        disbursement.disbursementScheduleStatus,
-        10,
-        SPACE_FILLER,
-      );
-      record.appendDate(disbursement.disbursementDate, DATE_FORMAT);
-      disbursement.dateSent
-        ? record.appendDate(disbursement.dateSent, DATE_FORMAT)
-        : record.repeatAppend(SPACE_FILLER, 8);
-      disbursement.disbursementValues.forEach(
-        (disbursementValue: DisbursementValue) => {
-          record.append(disbursementValue.valueCode, 4);
-          record.appendWithStartFiller(
-            round(disbursementValue.valueAmount),
-            8,
-            NUMBER_FILLER,
-          );
-        },
-      );
-      return record.toString();
+    const record = new StringBuilder();
+    record.appendWithStartFiller(this.assessmentId, 10, NUMBER_FILLER);
+    record.appendWithStartFiller(this.disbursementId, 10, NUMBER_FILLER);
+    record.append(this.applicationNumber, 10);
+    record.appendWithEndFiller(
+      emptyStringFiller(this.institutionStudentNumber),
```

Comment:
`emptyStringFiller` does not rely on the same const used on line 199.
I was under the impression that the `string-builder` methods would be adapted to accommodate this situation.